core/modules/wsched/ScanScheduler.h

```diff
@@ -82,7 +83,7 @@ class ScanScheduler : public SchedulerBase {
 
 private:
     bool _ready();
-    std::shared_ptr<ChunkDisk> _disk; //< Constrains access to files.
+    std::shared_ptr<ChunkTaskCollection> _disk; //< Constrains access to files.
```

Comment:
I'd rather you use a different name. The structure is important enough as it is. Using the same name for two different things makes difficult to keep the context in your head (well, actually impossible).